src/osd/PG.cc

```diff
@@ -1428,15 +1428,28 @@ bool PG::choose_acting(pg_shard_t &auth_log_shard_id,
     }
   }
 
-  // We go incomplete if below min_size for ec_pools since backfill
+  // We go incomplete if below data_chunk count for ec_pools since backfill
   // does not currently maintain rollbackability
   // Otherwise, we will go "peered", but not "active"
-  if (num_want_acting < pool.info.min_size &&
-      (pool.info.ec_pool() ||
-       !cct->_conf->osd_allow_recovery_below_min_size)) {
-    want_acting.clear();
-    dout(10) << "choose_acting failed, below min size" << dendl;
-    return false;
+  if (num_want_acting < pool.info.min_size) { 
+    bool ret = true;
```

Comment:
The name ```ret``` is very confusing here. We return if ret is *false*, and if ret is true we just fall through and might or might not succeed later.

How about we call it "must_fail" or something else more descriptive?